tests/cpp/data/test_sparse_page_dmatrix.cc

```diff
@@ -19,33 +18,34 @@ TEST(SparsePageDMatrix, MetaInfo) {
   EXPECT_EQ(dmat->Info().labels_.Size(), dmat->Info().num_row_);
 
   // Clean up of external memory files
+  std::remove(tmp_file.c_str());
   std::remove((tmp_file + ".cache").c_str());
   std::remove((tmp_file + ".cache.row.page").c_str());
 
   delete dmat;
 }
 
 TEST(SparsePageDMatrix, RowAccess) {
-  std::string tmp_file = CreateSimpleTestData();
+  // Create sufficiently large data to make two row pages
+  std::string tmp_file = CreateBigTestData(5000000);
   xgboost::DMatrix * dmat = xgboost::DMatrix::Load(
     tmp_file + "#" + tmp_file + ".cache", true, false);
   std::remove(tmp_file.c_str());
   EXPECT_TRUE(FileExists(tmp_file + ".cache.row.page"));
 
-  auto row_iter = dmat->RowIterator();
-  // Loop over the batches and count the records
+   //Loop over the batches and count the records
   long row_count = 0;
-  row_iter->BeforeFirst();
-  while (row_iter->Next()) row_count += row_iter->Value().Size();
+  for(auto &batch:dmat->GetRowBatches())
+  {
+    row_count += batch.Size();
+  }
   EXPECT_EQ(row_count, dmat->Info().num_row_);
   // Test the data read into the first row
-  row_iter->BeforeFirst();
-  row_iter->Next();
-  auto first_row = row_iter->Value()[0];
+  auto &batch = *dmat->GetRowBatches().begin();
```

Comment:
Add a test for non-emptiness of `dmat->GetRowBatches()`?

Reply:
I think emptiness will be caught by the existing tests.